test/foundry/Tiers.t.sol

```diff
@@ -0,0 +1,208 @@
+        vault.setERC20Tier(address(tier), fuzzedMinTier, fuzzedContext, fuzzedData);
+
+        // Stop the prank
+        vm.stopPrank();
+    }
+
+    /// Test setERC1155Tier event
+    function testSetERC1155Tier(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        bytes memory fuzzedData,
+        uint8 fuzzedMinTier,
+        uint256[] memory fuzzedContext,
+        address tier
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        fuzzedMinTier = uint8(bound(fuzzedMinTier, uint256(1), uint256(8)));
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+
+        // Create the vault
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Grant the necessary role
+        vault.grantRole(vault.ERC1155TIERER(), alice);
+
+        // Emit the expected event
+        vm.expectEmit(true, true, true, true);
+        emit SetERC1155Tier(alice, tier, fuzzedMinTier, fuzzedContext, fuzzedData);
+
+        // Call the function that should emit the event
+        vault.setERC1155Tier(tier, fuzzedMinTier, fuzzedContext, fuzzedData);
+
+        // Stop the prank
+        vm.stopPrank();
+    }
+
+    /// Test authorizeReceiptTransfer reverts if unauthorizedSenderTier
+    function testTransferOnUnauthorizedSenderTier(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        string memory assetName,
+        string memory assetSymbol,
+        bytes memory fuzzedData,
+        uint8 fuzzedMinTier,
+        uint256[] memory fuzzedContext,
+        uint256 certifyUntil,
+        uint256 referenceBlockNumber,
+        address tierAddress
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyBob = bound(fuzzedKeyBob, 1, SECP256K1_ORDER - 1);
+        address bob = vm.addr(fuzzedKeyBob);
+
+        referenceBlockNumber = bound(referenceBlockNumber, 1, block.number);
+        certifyUntil = bound(certifyUntil, 1, 1e6 - 1);
+
+        vm.assume(alice != bob);
+        vm.assume(tierAddress != address(0));
+
+        fuzzedMinTier = uint8(bound(fuzzedMinTier, uint256(1), uint256(8)));
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        vault.grantRole(vault.CERTIFIER(), alice);
+
+        // Call the certify function
+        vault.certify(certifyUntil, referenceBlockNumber, false, fuzzedData);
```

Comment:
should we be fuzzing a warp https://book.getfoundry.sh/cheatcodes/warp so that our certify is checked against times other than `0`?